### PR TITLE
DDCE-4692 - Update to use mongo cache lib

### PR DESCRIPTION
**Pull request by MRBeresford**
# DDCE-4692

### CODE REVIEW

File: app/controllers/OrganisationDetailsController.scala [hunk 1]
```diff
@@ -85,12 +85,11 @@ class OrganisationDetailsController @Inject()(
               )
             },
             data => {
-              shortLivedCache.cache[OrganisationDetails](cacheId, OrganisationDetails.cacheKey, data).flatMap { _ =>
+              sessionCacheRepository.putSession[OrganisationDetails](DataKey(OrganisationDetails.cacheKey), data).flatMap { _ =>
                 logger.debug(s"BusinessStructure retrieved: ${businessStructure.businessStructure}")
                 rosmService.rosmRegister(businessStructure, data).flatMap {
-                  case Right(safeId) =>
-                    logger.debug("rosmRegister Successful")
```
Comment (ina-tu) [inline app/controllers/OrganisationDetailsController.scala:92]: Did you mean to remove the `logger.debug` line? I think it's still useful to have

File: app/repositories/LisaCacheRepository.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmrc.mongo.cache.{CacheItem, SessionCacheRepository}
+import uk.gov.hmrc.mongo.{CurrentTimestampSupport, MongoComponent}
+
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class LisaCacheRepository @Inject()(
```
Comment (ina-tu) [inline app/repositories/LisaCacheRepository.scala:31]: missing test spec for LisaCacheRepository

**Review by ina-tu — CHANGES_REQUESTED** [review]
The acceptance tests are failing at [this](https://github.com/hmrc/lisa-api-tests/blob/33f956e2f1a4414517a30c234afb9ac2632ad022/src/test/resources/features/ui/registerOrganisation.feature#L936) scenario. I tested it locally and it doesn't redirect to the next page.

File: app/repositories/LisaCacheRepository.scala (hunk 2 above)
Comment (MRBeresford) [reply app/repositories/LisaCacheRepository.scala:31]: I dont think adding tests for the LisaCacheRepository would be helpful. 
This class extends `SessionCacheRepository` which I assume is already tested and dosnt have much other functionality to test.

**Comment by MRBeresford** [comment]
@ina-tu to answer your points:
- I fixed the failing acceptance test
- the info does not need to be encrypted 
- Ive added back in the log line (didn't meant to delete it good spot!)

File: app/controllers/OrganisationDetailsController.scala (hunk 1 above)
Comment (MRBeresford) [reply app/controllers/OrganisationDetailsController.scala:92]: ill add it back in 👍

File: app/repositories/LisaCacheRepository.scala (hunk 2 above)
Comment (ina-tu) [reply app/repositories/LisaCacheRepository.scala:31]: True, `SessionCacheRepository` is already tested.

You could add a test for the function `getFullCache` to make sure it behaves as expected.

**Review by ina-tu — APPROVED** [review]

**Review by ina-tu — APPROVED** [review]

**Pull request merged**